be/src/storage/lake/persistent_index_memtable.h

```diff
@@ -43,14 +44,23 @@ class PersistentIndexMemtable {
     Status get(size_t n, const Slice* keys, IndexValue* values, KeyIndexesInfo* not_found, size_t* num_found,
                int64_t version);
 
+    // |version|: version of index values
+    Status get(size_t n, const Slice* keys, IndexValue* values, KeyIndexesInfo* keys_info,
+               KeyIndexesInfo* found_keys_info, int64_t version);
+
+    size_t memory_usage();
+
+    Status flush(WritableFile* wf, uint64_t* filesz);
+
     void clear();
 
 private:
-    static void update_index_value(std::list<IndexValueInfo>* index_value_info, int64_t version,
+    static void update_index_value(std::list<IndexValueWithVer>* index_value_info, int64_t version,
                                    const IndexValue& value);
 
 private:
-    phmap::btree_map<std::string, std::list<IndexValueInfo>, std::less<>> _map;
+    phmap::btree_map<std::string, std::list<IndexValueWithVer>, std::less<>> _map;
```

Comment:
what's the expected size range of this _map? if the number of elements is relatively small (e.g. < 1k), should consider use unordered_map for a friendly gdb debugging. It's nearly impossible to debug a phmap in gdb.

Reply:
We use phmap for performance and order. When we write to sstable, the elements should be kept in order.

Reply:
it's fine for perf. however it doesn't make sense to use a complex structure if only expects a small number of elements in the map, right? In that case the human readable will overrule the perf, unless it is proved that under a such small size, the perf impact will be significant. Leave a comment about the size expectation or perf requirement in case the phmap is the only choice here.

Reply:
The default size of memtable is 52428800. If the size of key is 100, and the sizeof IndexValueWithVer is around 128. The number of elements will be 229950.